### PR TITLE
feature: allows generate::Doc to accept inputSpec as a directory

### PR DESCRIPTION
**Pull request by sophiejanssens**

### CODE REVIEW

File: boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.io.File;
+
+@Slf4j
+public class GenerateFromDirectoryDocMojo extends GenerateMojo {
+  // allows generate::Doc to accept inputSpec as a directory
+  // output docs will be places in deperate folders for each spec
```
Comment (torrespro) [inline boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:11]: typo: separate

File: boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      inputSpecs = inputSpecFile.listFiles(pathname -> pathname.getName().endsWith(".yaml"));
+
+      if (inputSpecs == null || inputSpecs.length == 0) {
+        throw new MojoExecutionException("No OpenAPI specs found in: " + inputSpec);
+      }
+
+      for(File f : inputSpecs){
+        inputSpec = f.getPath();
+        output = new File(outPutDirectory.getPath()+"/"+f.getName().substring(0,f.getName().lastIndexOf(".")).concat("-docs"));
```
Comment (jjjasper) [inline boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:31]: Hardcoded '/' might break on windows.

File: boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.backbase.oss.boat;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.plugin.MojoExecutionException;
+
+import java.io.File;
+
+@Slf4j
+public class GenerateFromDirectoryDocMojo extends GenerateMojo {
+  // allows generate::Doc to accept inputSpec as a directory
```
Comment (christopherm-backbase) [inline boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:10]: This could be javadoc.

Comment (sophiejanssens) [reply boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java]: Fixed

**Pull request merged**